src/ca350.py

```diff
@@ -847,6 +847,7 @@ def send_autodiscover(name, entity_id, entity_type, state_topic = None, device_c
         
     if unit_of_measurement:
         discovery_message["unit_of_measurement"] = unit_of_measurement
+        discovery_message["state_class"] = "measurement"
```

Comment:
This would cause all entities to carry the state_class=measurement property, which is not correct for e.g. the summer_mode sensor and non-sensor entities. I'd rather see this implemented the same way as `device_class` property below, and added to the respective `send_autodiscover` function calls.

Reply:
Fully agree! @zanac if you don't know how to do it I can do it later today or tomorrow.